### PR TITLE
Re-publish all warnings on BSP server startup

### PR DESCRIPTION
**Pull request by jvican**
... and use the correct file URI schema for the files in the diagnostics notifications. More info in the commit messages.

### CODE REVIEW

File: frontend/src/main/scala/bloop/logging/BspServerLogger.scala [hunk 1]
```diff
@@ -101,7 +101,7 @@ final class BspServerLogger private (
   }
 
   def noDiagnostic(project: Project, file: File): Unit = {
-    val uri = bsp.Uri(file.toURI)
+    val uri = bsp.Uri(file.toPath.toUri)
```
Comment (olafurpg) [inline frontend/src/main/scala/bloop/logging/BspServerLogger.scala]: 👍

File: frontend/src/test/scala/bloop/bsp/BspProtocolSpec.scala [hunk 2]
```diff
@@ -395,6 +395,74 @@ class BspProtocolSpec {
...
+            Assert.assertTrue(
+              s"URI ${tid.uri} does not start with valid 'file:///'",
+              tid.uri.value.startsWith("file:///")
+            )
+            val relativePath = AbsolutePath(tid.uri.toPath).toRelative(baseDir)
+            val canonical = relativePath.toString.replace(File.separatorChar, '/')
+            val report = diagnostics
+              .map(_.toString.replace("\n", " ").replace(System.lineSeparator, " "))
+            builder
+              .++=(s"#${cycleIndex + 1}: $canonical\n")
```
Comment (olafurpg) [inline frontend/src/test/scala/bloop/bsp/BspProtocolSpec.scala]: nit I personally prefer `.append` over `++=` when using non-infix syntax (also makes it easier to refactor to `java.lang.StringBuilder`

Comment (jvican) [reply frontend/src/test/scala/bloop/bsp/BspProtocolSpec.scala]: I'll keep it in mind whenever I touch this code again :+1:

**Comment by jvican** [comment]
> I am a bit concerned by the growing test case in BspProtocolSpec
  
Agreed, I would like to take some time down the road to deduplicate some of the logic here.

**Pull request merged**